### ISSUE TITLE
toggle selection range with shift

### ISSUE THREAD

**Issue by vogler** [charmbracelet/gum#241]
**Is your feature request related to a problem? Please describe.**
Selecting n consecutive items requires 2n key presses.
Selecting all items should be possible with the sequence space, l, shift+space.

**Describe the solution you'd like**
If shift is held, toggle a range of items, similar to selecting multiple files with shift in Windows or macOS.
Assume it's a range from `a` to `b`, then you would toggle items `]a, b]`.
`b` is the item you're at when you press shift+space/tab.
`a` could be
1. the item you toggled last
2. the item you toggled closest to you

Instead of toggling, one could also select/deselect depending on the current item (might be less surprising if there are toggled items inbetween).

**Comment by maaslalani** [comment]
Question: what if <kbd>tab</kbd> selected the item and went to the next item. That way you could hold down tab to select a range of items. Would that solve this issue for you?

**Comment by vogler** [comment]
Not really, that would be dependent on the key repeat rate and hard to use precisely for large ranges.

**Comment by caarlos0** [comment]
Hey!

I get the key repeat thing, but IMHO that is simple enough and work in most cases... 

Choosing to not do this, at least for now.

Thanks!
